NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs

```diff
@@ -205,16 +223,77 @@ private async Task GetNewNationsAsync()
             }
         }
 
+        private async Task EnsurePoolFilled()
+        {
+            var id = LogEventIdProvider.GetEventIdByType(LoggingEvent.EnsurePoolFilled);
+            List<REGION> regionsToRecruitFrom = new List<REGION>();
+            var regionNames = _config.RegionsToRecruitFrom.Split(";");
+            foreach (var regionName in regionNames)
+            {
+                var region = await _dumpDataService.GetRegionAsync(regionName);
+                if (region != null)
+                {
+                    regionsToRecruitFrom.Add(region);
+                    _logger.LogDebug(id, LogMessageBuilder.Build(id, $"Region '{regionName}' added to regionsToRecruitFrom."));
+                }
+                else
+                {
+                    _logger.LogWarning(id, LogMessageBuilder.Build(id, $"Region for name '{regionName}' couldn't be found in dumps."));
+                }
+            }
+            while (IsRecruiting)
+            {
+                bool fillingUp = false;
+                int counter = 0;
+                while (NationManager.GetNationCountByStatusName("pending") < _config.MinimumRecruitmentPoolSize)
+                {
+                    if (!fillingUp)
+                    {
+                        fillingUp = true;
+                        _logger.LogInformation(id, LogMessageBuilder.Build(id, $"Filling up pending pool now from {NationManager.GetNationCountByStatusName("pending")} to {_config.MinimumRecruitmentPoolSize}"));
+                    }
+                    var regionId = _rnd.Next(regionsToRecruitFrom.Count);
+                    var region = regionsToRecruitFrom.ElementAt(regionId);
+                    string nationName;
+                    do
+                    {
+                        var nationId = _rnd.Next(region.NATIONNAMES.Count);
+                        nationName = region.NATIONNAMES.ElementAt(nationId);
+                    }
+                    while (await NationManager.IsNationPendingSkippedSendOrFailedAsync(nationName) || !await IsNationRecruitableAsync(nationName, id));
```

Comment:
Why is this while loop empty?

Reply:
It should find an nation to be added. So it isn't empty. As IsNationRecruitableAsync will take some time to complete this shouldn't get CPU temperature high.